src/host/ut_host/ScreenBufferTests.cpp

```diff
@@ -2502,6 +2506,60 @@ void ScreenBufferTests::SetColorTableThreeDigits()
 
 }
 
+
+void ScreenBufferTests::SetDefaultForegroundColor()
+{
+    // Setting the default foreground color should work
+
+    CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    gci.LockConsole(); // Lock must be taken to swap buffers.
+    auto unlock = wil::scope_exit([&] { gci.UnlockConsole(); });
+
+    SCREEN_INFORMATION& mainBuffer = gci.GetActiveOutputBuffer();
+    VERIFY_IS_FALSE(mainBuffer._IsAltBuffer());
+    WI_SetFlag(mainBuffer.OutputMode, ENABLE_VIRTUAL_TERMINAL_PROCESSING);
+    VERIFY_IS_TRUE(WI_IsFlagSet(mainBuffer.OutputMode, ENABLE_VIRTUAL_TERMINAL_PROCESSING));
+
+    StateMachine& stateMachine = mainBuffer.GetStateMachine();
+
+    const COLORREF originalColor = gci.GetDefaultForegroundColor();
+    const COLORREF testColor = RGB(0x33, 0x66, 0x99);
+    VERIFY_ARE_NOT_EQUAL(originalColor, testColor);
+
+    std::wstring seq = L"\x1b]10;rgb:33/66/99\x1b\\";
+    stateMachine.ProcessString(seq);
+
+    VERIFY_ARE_EQUAL(testColor, gci.GetDefaultForegroundColor());
+}
+
+
+void ScreenBufferTests::SetDefaultBackgroundColor()
+{
+    // Setting the default foreground color should work
+
+    CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    gci.LockConsole(); // Lock must be taken to swap buffers.
+    auto unlock = wil::scope_exit([&] { gci.UnlockConsole(); });
+
+    SCREEN_INFORMATION& mainBuffer = gci.GetActiveOutputBuffer();
+    VERIFY_IS_FALSE(mainBuffer._IsAltBuffer());
+    WI_SetFlag(mainBuffer.OutputMode, ENABLE_VIRTUAL_TERMINAL_PROCESSING);
+    VERIFY_IS_TRUE(WI_IsFlagSet(mainBuffer.OutputMode, ENABLE_VIRTUAL_TERMINAL_PROCESSING));
+
+    StateMachine& stateMachine = mainBuffer.GetStateMachine();
+
+    const COLORREF originalColor = gci.GetDefaultBackgroundColor();
+    const COLORREF testColor = RGB(0x33, 0x66, 0x99);

```

Comment:
just on a lark: can we test things that go outside the 0-255 or 0-ff (?) range? some negative cases and some cases for longer inputs might be interesting to see

Reply:
Sure, but the parser for _that_ isn't my code (I re-used the existing color parser).

Reply:
Added just a couple. How deep do you want to go? They're all just going to silently do nothing unless we find a bug in the parsers.

There should probably be (maybe there are) unit tests for the color parser already?